### PR TITLE
save transient layers in permalink

### PR DESCRIPTION
**Pull request by LukasLohoff**
Saves external layers in permalink and restores them when a permalink is opened.

Requires https://github.com/terrestris/ol-util/pull/796

@terrestris/devs Please review

Also adds a watchBuild script which allows to link a local shogun-gis-client into another package. Based on https://github.com/terrestris/watch-build-copy but customized for this repo.

### CODE REVIEW

**Review by annarieger — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -30,7 +33,8 @@
     "test": "jest --coverage",
     "test:ui": "playwright test",
     "tsc": "tsc -p tsconfig.json",
-    "typecheck": "tsc --noEmit -p tsconfig.json"
+    "typecheck": "tsc --noEmit -p tsconfig.json",
+    "watch:buildto": "node watchBuild.js"
```
Comment (dnlkoch) [inline package.json:37]: Why don't we use the [watch-copy-build](https://github.com/terrestris/watch-build-copy) here?

File: src/bootstrap.tsx [hunk 2]
```diff
@@ -286,6 +286,20 @@ const setupSHOGunMap = async (application: Application) => {
 
   view.setConstrainResolution(true);
 
+  const processLayerGroup = new OlLayerGroup({
```
Comment (dnlkoch) [inline src/bootstrap.tsx:289]: What is a `processLayer`? To me this seems to be project related? Why do we need to initialize it in the shogun-gis-client?

File: src/components/Permalink/index.tsx [hunk 3]
```diff
@@ -80,12 +103,17 @@ export const Permalink: React.FC<PermalinkProps> = () => {
 
       const listenerKeyCenter = map.getView().on('change:center', updatePermalink);
       const listenerKeyResolution = map.getView().on('change:resolution', updatePermalink);
+      const listenerLayerGroup = map.on('change:layergroup', () => {
+        // todo: test if permalink is updated correctly
```
Comment (dnlkoch) [inline src/components/Permalink/index.tsx:107]: Is this still a TODO?

File: src/components/ToolMenu/LayerTree/index.tsx [hunk 4]
```diff
@@ -46,12 +46,23 @@ export const LayerTree: React.FC<LayerTreeProps> = ({
   const [visibleLegendsIds, setVisibleLegendsIds] = useState<string[]>([]);
 
   const treeFilterFunction = (layer: OlLayer<OlSource> | OlLayerGroup) => {
+    if (layer instanceof OlLayerGroup && !layer.getVisible() && layer.getLayers().getLength() === 0) {
+      return false;
+    }
+
+    if (layer.get('name') === 'react-geo_digitize') {
```
Comment (dnlkoch) [inline src/components/ToolMenu/LayerTree/index.tsx:53]: Is this check needed? We check for an ol vector source later on.

**Pull request merged**